docs/tsm/TSM_FORMAT.md

```diff
@@ -0,0 +1,175 @@
+# TSM Inverted Index Proposal
+
+This is a proposal for adding a disk-based, inverted index for measurements, tags and values.  This is a extension and/or alternative of the ideas proposed in [TSI Proposal #7174](https://github.com/influxdata/influxdb/pull/7174).
+
+The inverted index supports the following:
+
+* Hash lookups of measurement, tags, and values
+* Sorted iteration over measurement, tags and values
+* Efficient series boolean operations over compressed bitmaps
+* Compressed series keys
+
+The index is a separate set of files specific to a shard that are compacted and merged periodically.  Searches run across multiple files and are merged.
+
+## Format
+
+```
+┌────────────────────────────────────────────────────────────────┐
+│                         Inverted Index                         │
+├────────────────────┬──────────────────┬────┬───────────────────┤
+│    Series Dict     │     Tag Sets     │... │   Measurements    │
+└────────────────────┴──────────────────┴────┴───────────────────┘
+```
+
+The `Inverted Index` contains three sections: Series Dictionary, Tag Sets and Measurements.  The `Series Dictionary` contains a dictionary encoded version of all series in the index.  The `Tag Sets` are the unique tag/value pairs for a given measurement.  The `Measurements` are the unique set of measurements in the index.
+
+```
+┌───────────────────────────────────────────────────────┐
+│                   Series Dictionary                   │
+├───────────────────────────┬───────────────────────────┤
+│        Dictionary         │          Series           │
+├─────────┬─────────────────┼─────────┬─────────────────┤
+│   Len   │  Sorted Terms   │   Len   │  Sorted Series  │
+│ 4 bytes │                 │ 4 bytes │                 │
+└─────────┴─────────────────┴─────────┴─────────────────┘
+```
+
+The `Series Dictionary` contains two sections.  The first is an array of unique terms for all series keys.  For example, `cpu,host=server0`, would create an array `3cpu4host7server0` where each term is prefixed with it's variable byte encoded length.
```

Comment:
Possibly.  I was thinking that measurement and tag keys are in the dictionary before tag values because they would be more commonly seen.  Having them earlier in the dictionary increases their chance of have a single (or double) byte representation.

There are other dictionary encoding schemes we could look at too.  I only added this just for discussion on whether compressing the dictionary would be a good idea or not.